chat-plugins/othermetas.js

```diff
@@ -0,0 +1,98 @@
+//Other metas plugin by Spandan
+'use strict';
+
+exports.commands = {
+	mixandmega: 'mnm',
+	mnm: function (target, room, user) {
+		if (!this.runBroadcast()) return;
+		if (!target || toId(target) === "" || !target.includes('@')) return this.parse('/help mixandmega');
+		let sep = target.split('@');
+		let stone = toId(sep[1]), template = toId(sep[0]);
+		if ((!Tools.data.Items[stone] || !Tools.data.Items[stone].megaEvolves) && !Tools.data.Items[stone].onPrimal) {
+			return this.errorReply('Error: Mega Stone not found');
+		}
+		if (!Tools.data.Pokedex[toId(template)]) {
+			return this.errorReply("Error: Pokemon not found");
+		}
+		template = Object.assign({}, Tools.getTemplate(template));
+		stone = Object.assign({}, Tools.getItem(stone));
+		if (template.isMega || (template.evos && Object.keys(template.evos).length > 0)) {
+			return this.errorReply(`You cannot mega evolve ${template.name} in Mix and Mega.`);
+		}
+		let deltas; //Get mega deltas.
+		let baseTemplate = Tools.getTemplate(stone.megaEvolves), megaTemplate = Tools.getTemplate(stone.megaStone);
+		if (stone.id === 'redorb') {
+			megaTemplate = Tools.getTemplate("Groudon-Primal");
+			baseTemplate = Tools.getTemplate("Groudon");
+		} else if (stone.id === 'blueorb') {
+			megaTemplate = Tools.getTemplate("Kyogre-Primal");
+			baseTemplate = Tools.getTemplate("Kyogre");
+		}
+		deltas = {
+			ability: megaTemplate.abilities['0'],
+			baseStats: {},
+			weightkg: megaTemplate.weightkg - baseTemplate.weightkg,
+		};
+		for (let statId in megaTemplate.baseStats) {
+			deltas.baseStats[statId] = megaTemplate.baseStats[statId] - baseTemplate.baseStats[statId];
+		}
+		if (megaTemplate.types.length > baseTemplate.types.length) {
+			deltas.type = megaTemplate.types[1];
+		} else if (megaTemplate.types.length < baseTemplate.types.length) {
+			deltas.type = baseTemplate.types[0];
+		} else if (megaTemplate.types[1] !== baseTemplate.types[1]) {
+			deltas.type = megaTemplate.types[1];
+		}
+		//////////////////////////////////////////
+		let ability = deltas.ability, types = template.types, baseStats = Object.assign({}, template.baseStats);
+		if (types[0] === deltas.type) {
+			types = [deltas.type];
+		} else if (deltas.type) {
+			types = [types[0], deltas.type];
+		}
+		for (let statName in baseStats) {
+			baseStats[statName] = Tools.clampIntRange(baseStats[statName] + deltas.baseStats[statName], 1, 255);
+		}
+		let weightkg = Math.max(0.1, template.weightkg + deltas.weightkg);
```

Comment:
`Math.round()` removes **any** decimal places, so I thought I'd not do that... 
I could:
`weightkg = Math.round(Math.max(0.1, template.weightkg + deltas.weightkg)*100)/100`;
Which would keep two decimal places.
What do you think about it?